### PR TITLE
Update plugins installer component to TS, Fix TS bugs and Syncpack TypeScript

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

After upgrading to pnpm 7, intermittent errors were occuring on build:

```
woocommerce/client/admin:turbo:build: ERROR in client/tasks/fills/tax/woocommerce-tax/plugins.tsx:77:5
woocommerce/client/admin:turbo:build: TS2322: Type '{ onComplete: (activatedPlugins: string[], response: InstallPluginsResponse) => void; onError: (errors: unknown, response: unknown) => void; onSkip: () => void; skipText: string; onAbort: () => void; abortText: string; }' is not assignable to type 'IntrinsicAttributes & { children?: ReactNode; }'.
woocommerce/client/admin:turbo:build:   Property 'onComplete' does not exist on type 'IntrinsicAttributes & { children?: ReactNode; }'.
```

This PR updates the `Plugins` installer component to a functional TS component with respective types to fix the issue. It also adjusts the tests to match the fact that we can't mock `installAndActivatePlugins` in the same way now its declared inside the component.

Additionally this fixes some other small TS issues that arose and matches TypeScript version across repo with Syncpack so that every project is compiled with the same version.

### How to test the changes in this Pull Request:

1. Check out this branch
2. Run `pnpm i && pnpm run build`
3. Smoke test all uses of the plugin component (taxes task, experimental shipping task, payments task, WCPay task, etc)
4. Check that the plugins are install/activated correctly
5. Deactivate plugins and attempt other actions in the component, such as skipping (note that skip text may be different in each component)

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/src/plugins/index.tsx
Comment (samueljseay) [inline packages/js/components/src/plugins/index.tsx:21]: Changed this from `onError` because you declare a default value it is never null.

File: packages/js/components/src/plugins/index.tsx
Comment (samueljseay) [inline packages/js/components/src/plugins/index.tsx:36]: onSkip should not have default value, it changed the runtime behaviour and  broke tests.

File: packages/js/components/src/plugins/test/index.js
Comment (samueljseay) [inline packages/js/components/src/plugins/test/index.js:15]: had to adjust mocking for useSelect and useDispatch now there is no HOC to pass install function to.

File: packages/js/components/src/plugins/test/index.js
Comment (samueljseay) [inline packages/js/components/src/plugins/test/index.js:119]: no behaviour change in test here just adjusting for new mocking

**Comment by samueljseay** [comment]
@jonathansadowski wonder if you could give this one a review, I think its fairly important for making the build process and pnpm upgrade more reliable and I've noticed it seems to fix a lot of issues.

File: .syncpackrc
Comment (jonathansadowski) [inline .syncpackrc:30]: We should add a versionGroup for `typescript` here as well.

Reasoning: if we ever need to go backwards in version number, syncpack will not be able to do this for us absent having a version group defined. It also helps with consistency and having a quick glanceable look at all the versions that are managed by syncpack.

**Review by jonathansadowski — CHANGES_REQUESTED** [review]

File: .syncpackrc
Comment (samueljseay) [reply .syncpackrc:30]: @jonathansadowski sounds good! I'll add that

Comment (samueljseay) [reply .syncpackrc:30]: Resolved in https://github.com/woocommerce/woocommerce/pull/34787/commits/6033e6778b3e3377b87df7a53deb550ba8430e00

**Review by jonathansadowski — APPROVED** [review]
The syncpack changes in this PR look good to me. I verified that I was able to install and build without issue. I also attempted to follow the remainder of the testing instructions, and while nothing looked amiss to me, I'm admittedly not as familiar with those components, so you may want to wait for another review there.

**Review by louwie17 — APPROVED** [review]
Changes look good and this tested well, thanks for all your work on this @joshuatf & @samueljseay 🌮 🎉

**Pull request merged**
